core/src/util/decimal128.rs

```diff
@@ -0,0 +1,466 @@
+use likely_stable::unlikely;
```

Comment:
Up to you whether you want to read this file. It's mostly just a translation of the C code, and it works, so we know the translation was correct haha. If you want to suggest some better rustisms I'd be up for it.